byway/src/main/webapp/track-interests.js

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+let selected = [];
+
+/** 
+ * Track the selected interests in an array according to their status. 
+ * @param elem tracks the current button element chosen.
+ */
+function checkStatus(elem) {
+  if(elem.className === "btnActive") {
+    selected.push(elem.innerText);
+  } else {
+    selected = selected.filter(function(interest) {
+      return interest !== elem.innerText;
+    });
+  }
+}
+
+/** 
+ * Update the class of the buttons to change their style.
+ * @param elem tracks the current button element chosen.
+ */
+function switchStatus(elem) {
+  let switchClass = (elem.className === "btn"? "btnActive": "btn");
+  elem.className = switchClass;
+  checkStatus(elem);
+}
+
+/** Display all the buttons onscreen with independent onClick events. */
+function loadButtons() {
+  fetch('/places')
+  .then(response => response.json())
+  .then((places) => {
+    let buttonSection = document.getElementById("interests");
+    places.forEach((place) => {
+      let button = document.createElement("button");
+      button.innerText = place;
+      button.setAttribute("onClick", "switchStatus(this)");
```

Comment:
To add listeners on elements, you should probably use addEventListener. Setting `onclick = <some function>` is OK as well (though functions differently), but either way setting the 'onClick' attribute to a string value is a bad idea, since then it's harder for automatic tools to check that the function you provided is correct (since the tools see it as a string) and you have much less control over what the function is doing.

Note that if you use `onclick = <some function>` or `addEventListener`, you can use a lambda expression to allow you to explicitly capture the affected `button`.

For the difference between `onclick` and `addEventListener`, see here:
https://stackoverflow.com/a/6348597

Reply:
I tried to add the event listener this way, but it doesn't seem to work: `button.addEventListener("cllick", () => switchStatus(this))`.  `this` seems to be passing in the entire window rather than the button element itself. How should I explicitly reference the button clicked using this event listener?

Reply:
I found a solution to properly reference the button element clicked. Fixed

Reply:
`() =>` functions exist specifically to avoid weird issues with `this` -- specifically they have no "this" of their own, so they take `this` from the outer function. But since you're writing an arrow function inside an outer function you can capture state from the surrounding function: `button.addEventListener("cllick", () => switchStatus(button))` should work fine and is more explicit than using some automatic `this` behavior.